src/ui/EditorNS/editor.cpp

```diff
@@ -268,117 +305,184 @@ namespace EditorNS
         sendMessage("C_CMD_MARK_DIRTY");
     }
 
-    QList<QMap<QString, QString>> Editor::languages()
+    void Editor::initLanguageCache()
     {
-        // TODO: Cache this data.
-        QJSEngine engine;
         QFileInfo fileInfo(Notepadqq::editorPath());
-        QString fileName = fileInfo.absolutePath() + "/classes/Languages.js";
+        QString fileName = fileInfo.absolutePath() + "/classes/Languages.json";
```

Comment:
Languages.json isn't a class anymore: it should probably be in `fileInfo.absolutePath() + "/Languages.json";`